### PR TITLE
PRODESC-5780: ACS Repo DAU APIs to also use non-attach allow list

### PR DESCRIPTION
**Pull request by montgolfiere**
- moved existing pre-configured allow list from remote-api to repository layer ("nodes.nonAttachContentTypes" xml -> "content.nonAttach.mimetypes" prop)
- now also used by DAU (as well as existing V1 REST API and CMIS to get/download content)

### CODE REVIEW

File: remote-api/src/main/java/org/alfresco/opencmis/CMISServletDispatcher.java [hunk 1]
```diff
@@ -133,9 +135,12 @@ public void setCmisVersion(String cmisVersion)
         this.cmisVersion = CmisVersion.fromValue(cmisVersion);
     }
 
-    public void setNonAttachContentTypes(Set<String> nonAttachWhiteList)
+    public void setNonAttachContentTypes(String nonAttachAllowListStr)
     {
-        this.nonAttachContentTypes = nonAttachWhiteList;
+		if (nonAttachAllowListStr != null) 
+		{
+			nonAttachContentTypes = Stream.of(nonAttachAllowListStr.trim().split("\\s*,\\s*")).collect(Collectors.toSet());
```
Comment (mpichura) [inline remote-api/src/main/java/org/alfresco/opencmis/CMISServletDispatcher.java:142]: Wouldn't following be a bit simpler?
```suggestion
			nonAttachContentTypes = Set.of(nonAttachAllowListStr.split(","));
```

File: remote-api/src/main/java/org/alfresco/rest/api/impl/NodesImpl.java [hunk 2]
```diff
@@ -241,9 +243,12 @@
 
     private Set<String> nonAttachContentTypes = Collections.emptySet(); // pre-configured whitelist, eg. images & pdf
 
-    public void setNonAttachContentTypes(Set<String> nonAttachWhiteList)
+    public void setNonAttachContentTypes(String nonAttachAllowListStr)
     {
-        this.nonAttachContentTypes = nonAttachWhiteList;
+        if (nonAttachAllowListStr != null)
+        {
+            nonAttachContentTypes = Stream.of(nonAttachAllowListStr.trim().split("\\s*,\\s*")).collect(Collectors.toSet());
```
Comment (mpichura) [inline remote-api/src/main/java/org/alfresco/rest/api/impl/NodesImpl.java:250]: Wouldn't following be a bit simpler?
```suggestion
            nonAttachContentTypes = Set.of(nonAttachAllowListStr.split(","));
```

**Review by mpichura — APPROVED** [review]
LGTM - just one minor remark which doesn't affect the functionality.

File: remote-api/src/main/java/org/alfresco/opencmis/CMISServletDispatcher.java (hunk 1 above)
Comment (PunchedChimera) [reply remote-api/src/main/java/org/alfresco/opencmis/CMISServletDispatcher.java:142]: `.split("\\s*,\\s*")` Is to remove trailing and leading whitespace? Probably needs to stay, makes suggestion:
```suggestion
			nonAttachContentTypes = Set.of(nonAttachAllowListStr.split("\\s*,\\s*"));
```

File: repository/src/main/java/org/alfresco/repo/content/ContentServiceImpl.java [hunk 3]
```diff
@@ -151,6 +157,14 @@ public void setSystemWideDirectUrlConfig(SystemWideDirectUrlConfig systemWideDir
         this.systemWideDirectUrlConfig = systemWideDirectUrlConfig;
     }
 
+    public void setNonAttachContentTypes(String nonAttachAllowListStr) 
+    {
+        if (nonAttachAllowListStr != null)
+        {
+            nonAttachContentTypes = Stream.of(nonAttachAllowListStr.trim().split("\\s*,\\s*")).collect(Collectors.toSet());
```
Comment (PunchedChimera) [inline repository/src/main/java/org/alfresco/repo/content/ContentServiceImpl.java:164]: As above, feedback from mpichura.

File: repository/src/main/java/org/alfresco/repo/content/ContentServiceImpl.java [hunk 4]
```diff
@@ -691,4 +706,21 @@ private Long adjustValidFor(Long validFor)
...
+        boolean attachment = true;
+        if (! attachmentIn)
+        {
+            if ((nonAttachContentTypes != null) && (nonAttachContentTypes.contains(mimeType)))
+            {
+                attachment = false;
+            }
+            else
+            {
+                logger.warn("Ignored attachment=false for " + nodeRef.getId() + " since " + mimeType + " is not in the whitelist for non-attach content types");
```
Comment (PunchedChimera) [inline repository/src/main/java/org/alfresco/repo/content/ContentServiceImpl.java:721]: allow-list, over whitelist? 😉

**Review by PunchedChimera — APPROVED** [review]
Some minor points. Otherwise it's good 👍

File: remote-api/src/main/java/org/alfresco/opencmis/CMISServletDispatcher.java (hunk 1 above)
Comment (montgolfiere) [reply remote-api/src/main/java/org/alfresco/opencmis/CMISServletDispatcher.java:142]: I did look at a few options. There are other parts of code base that use Google Commons (which I discounted).

As per @PunchedChimera it was related to ensuring we trim (potential) whitespace either side of comma as well as start and end of string. I realise that it is unlikely that this config will be overridden other than maybe option to have an empty list.

Comment (montgolfiere) [reply remote-api/src/main/java/org/alfresco/opencmis/CMISServletDispatcher.java]: I've updated based on your feedback (+ extra tweak). Thanks both 👍

File: repository/src/main/java/org/alfresco/repo/content/ContentServiceImpl.java (hunk 3 above)
Comment (montgolfiere) [reply repository/src/main/java/org/alfresco/repo/content/ContentServiceImpl.java]: I've updated based on your feedback (+ extra tweak). Thanks both 👍

File: remote-api/src/main/java/org/alfresco/rest/api/impl/NodesImpl.java (hunk 2 above)
Comment (montgolfiere) [reply remote-api/src/main/java/org/alfresco/rest/api/impl/NodesImpl.java]: Thanks - done (with tweaks) - see also earlier comment.

**Review by KikoUK — APPROVED** [review]

**Pull request merged**